crates/core/tedge_api/src/entity_store.rs

```diff
@@ -121,6 +130,8 @@ pub struct EntityStore {
     entity_id_index: HashMap<EntityExternalId, EntityTopicId>,
     external_id_mapper: ExternalIdMapperFn,
     external_id_validator_fn: ExternalIdValidatorFn,
+    // TODO: this is a c8y cloud specific concern and it'd be better to put it somewhere else.
+    default_service_type: String,
```

Comment:
I don't even get the purpose of default service type.

This should be related to the [inventory](https://github.com/thin-edge/thin-edge.io/issues/2279)

Reply:
Yes, currently the entity store kind of stores data relevant both to the local thin-edge instance but also to the cloud, i.e. entity external id, and I had to add the default service type because the local registration mainly happens inside the entity store code, and the store needs to be a single source of truth about entities and their properties.

[In another comment](https://github.com/thin-edge/thin-edge.io/pull/2310#discussion_r1349992273) I thought about making an entity store generic over additional data related to the cloud, so that we can use different types for data we always expect to be present for an entity when connected to a certain cloud, i.e. an external id.

I was not familiar with inventory proposal, so I'll have to take a look at it to know if we're not duplicating some of the behaviour that we could attach to an entity store.